setup_ctp7.sh

```diff
@@ -197,94 +198,109 @@ then
-        echo "Please verify that the specified tag exists"
-        exit
-    fi
+    pushd ${ctp7host}
+    find . -type d -print0 -exec chmod a+rx {} \+
+    find . -type f -print0 -exec chmod a+r  {} \+
+    find bin -type f -print0 -exec chmod a+rx {} \+
+    find lib -type f -print0 -exec chmod a+rx {} \+
 
-    echo "wget https://github.com/cms-gem-daq-project/xhal/releases/download/${xhaltag}/reg_interface.zip && unzip reg_interface.zip && rm reg_interface.zip"
-    wget https://github.com/cms-gem-daq-project/xhal/releases/download/${xhaltag}/reg_interface.zip
-    unzip reg_interface.zip -d python/
-    rm reg_interface.zip
-fi
+    ## Take latest versions
+    tarballs=(
+        ctp7-base.tgz     ## ipbus, liblmdb.so
+        reedmuller.tgz    ## libreedmuller.so, rmencode, rmdecode
+        rwreg.tgz         ## librwreg.so
+        reg_interface.tgz ## reg_interface
+        xhal.tgz          ## libxhal.so, reg_interface_gem
+        ctp7_modules.tgz  ## ctp7 RPC modules
+    )
+    set -x
+    for tb in ${tarballs[@]}
+    do
+        curl -L -O ${GEMDAQ_DOWNLOAD_URL}/${tb}
+        tar xzf ${tb}
+        rm -rf ${tb}
+    done
 
-# Update CTP7 gemdaq paths
-if [ -n "${update}" ]
-then
-    echo "Creating/updating CTP7 gemdaq directory structure"
-    echo "ssh root@${ctp7host} mkdir -p /mnt/persistent/gemdaq"
-    echo "ssh root@${ctp7host} mkdir -p /mnt/persistent/gemdaq/address_table.mdb"
-    echo "ssh root@${ctp7host} touch /mnt/persistent/gemdaq/address_table.mdb/data.mdb"
-    echo "ssh root@${ctp7host} touch /mnt/persistent/gemdaq/address_table.mdb/lock.mdb"
-    echo "ssh root@${ctp7host} chmod -R 777 /mnt/persistent/gemdaq/address_table.mdb"
-    ssh root@${ctp7host} mkdir -p /mnt/persistent/gemdaq && mkdir -p /mnt/persistent/gemdaq/address_table.mdb && touch /mnt/persistent/gemdaq/address_table.mdb/data.mdb && touch /mnt/persistent/gemdaq/address_table.mdb/lock.mdb && chmod -R 777 /mnt/persistent/gemdaq/address_table.mdb
+    ## Override if a specific version is specified
+    if [ -n "${xhaltag}" ]
+    then
+        curl -L ${GEMDAQ_DOWNLOAD_URL}/xhal-${xhaltag}.tgz -o xhal.tgz
+        tar xzf xhal.tgz
+        rm -rf xhal.zip
+    fi
 
-    find . -type d -print0 | xargs -0 -n1 chmod a+rx
-    find . -type f -print0 | xargs -0 -n1 chmod a+r
-    find bin -type f -print0 | xargs -0 -n1 chmod a+rx
-    find lib -type f -print0 | xargs -0 -n1 chmod a+rx
+    if [ -n "${ctp7modtag}" ]
+    then
+        curl -L ${GEMDAQ_DOWNLOAD_URL}/ctp7_modules-${ctp7modtag}.tgz -o ctp7_modules.tgz
+        tar xzf ctp7_modules.tgz
+        rm -rf ctp7_modules.tgz
+    fi
+    set +x
 
-    echo "wget https://raw.githubusercontent.com/cms-gem-daq-project/ctp7_modules/develop/conf/conf.txt"
-    wget https://raw.githubusercontent.com/cms-gem-daq-project/ctp7_modules/develop/conf/conf.txt
-    echo "cp conf.txt vfat3/conf.txt"
-    cp conf.txt vfat3/conf.txt
-    echo "rm -rf conf.txt"
+    ## Obsolete?
+    set -x
+    mkdir -p vfat3
```

Comment:
`setLinks.py` script is also required under `vfat3`

Reply:
is there a technical reason it can't it be used from its current location, i.e., `scripts`?

Reply:
I think that's where the `testConnectivitiy.py` (or one of the scripts it calls) will look for it.